### PR TITLE
feat: add support for standalone mode in Python SDK

### LINKED ISSUE #1279: Support standalone mode in Python SDK
Now Python SDK supports OpenMLB cluster mode and we should support standalone mode as well.

### PR DESCRIPTION
**Pull request by HuilinWu2**
now users are able to use standalone mode in python SDK DBAPI through:
```
import openmldb
# cluster mode
db = openmldb.dbapi.connect("db_test","127.0.0.1:6181","/onebox")

# standalone mode
db = openmldb.dbapi.connect("db_test","127.0.0.1", 6527)
```
or through sqlalchemy API:

```
import sqlalchemy as db
# cluster
engine = db.create_engine("openmldb:///db_test?zk=127.0.0.1:6181&zkPath=/onebox")

# standalone
engine = db.create_engine("openmldb:///db_test?host=127.0.0.1&port=6527")
```

### CODE REVIEW

File: python/openmldb/dbapi/dbapi.py [hunk 1]
```diff
@@ -505,8 +506,8 @@ def close(self):
         raise NotSupportedError("Unsupported in OpenMLDB")
 
     def cursor(self):
-        return Cursor(self._db, self._zk, self._zkPath, self)
+        return Cursor(self._db, self)
 
 # Constructor for creating connection to db
-def connect(db, zk, zkPath):
-    return Connection(db, zk, zkPath)
+def connect(db,*args,is_cluster_mode=True):
```
Comment (tobegit3hub) [inline python/openmldb/dbapi/dbapi.py]: Format the code and add spaces between the parameters.

File: python/openmldb/sdk/sdk.py [hunk 2]
```diff
@@ -25,27 +25,45 @@
...
+        if self.is_cluster_mode:
+            options = sql_router_sdk.SQLRouterOptions()
+            options.zk_cluster = self.options.zk_cluster
+            options.zk_path = self.options.zk_path
+            self.sdk = sql_router_sdk.NewClusterSQLRouter(options)
+            if not self.sdk:
+                logger.error("fail to init openmldb sdk with zk cluster %s and zk path %s"%(options.zk_cluster, options.zk_path))
+                return False
+            logger.info("init openmldb sdk done with zk cluster %s and zk path %s"%(options.zk_cluster, options.zk_path))
+            return True
```
Comment (tobegit3hub) [inline python/openmldb/sdk/sdk.py]: `return True` can be reused for both modes.

**Review by tobegit3hub — CHANGES_REQUESTED** [review]

File: python/openmldb/dbapi/dbapi.py [hunk 3]
```diff
@@ -456,12 +454,15 @@ def get_resultset_schema(self):
 
 class Connection(object):
 
-    def __init__(self, db, zk, zkPath):
+    def __init__(self, db,is_cluster_mode=True, *args):
```
Comment (tobegit3hub) [inline python/openmldb/dbapi/dbapi.py]: Format the code and add spaces between parameters.

File: python/openmldb/dbapi/dbapi.py [hunk 4]
```diff
@@ -456,12 +454,15 @@ def get_resultset_schema(self):
...
         self._connected = True
         self._db = db
-        self._zk = zk
-        self._zkPath = zkPath
-        options = sdk_module.OpenmldbSdkOptions(zk, zkPath)
+        self.zk_or_host = args[0]
```
Comment (tobegit3hub) [inline python/openmldb/dbapi/dbapi.py]: Remove `this` and do not set as object members.

It is not used and maybe delete it.

File: python/openmldb/dbapi/dbapi.py (hunk 1 above)
Comment (HuilinWu2) [reply python/openmldb/dbapi/dbapi.py]: updated.

File: python/openmldb/sdk/sdk.py (hunk 2 above)
Comment (HuilinWu2) [reply python/openmldb/sdk/sdk.py]: updated.

File: python/openmldb/dbapi/dbapi.py (hunk 3 above)
Comment (HuilinWu2) [reply python/openmldb/dbapi/dbapi.py]: updated.

File: python/openmldb/dbapi/dbapi.py (hunk 4 above)
Comment (HuilinWu2) [reply python/openmldb/dbapi/dbapi.py]: updated.

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Comment by dl239** [comment]
> now users are able to use standalone mode in python SDK DBAPI through:
> 
> ```
> import openmldb
> # cluster mode
> db = openmldb.dbapi.connect("db_test","127.0.0.1:6181","/onebox")
> 
> # standalone mode
> db = openmldb.dbapi.connect("db_test", host="127.0.0.1", port=6527, is_cluster_mode=False)
> ```
> 
> or through sqlalchemy API:
> 
> ```
> import sqlalchemy as db
> # cluster
> engine = db.create_engine("openmldb:///db_test?zk=127.0.0.1:6181&zkPath=/onebox")
> 
> # standalone
> engine = db.create_engine("openmldb:///db_test?host=127.0.0.1&port=6527", connect_args={'is_cluster_mode':False})
> ```

why add `is_cluster_mode` option to host and port connect method. It's better to keep consist with CLI that zk as cluster and host/port as standalone

File: python/openmldb/test/sql_smoke_test.py [hunk 5]
```diff
@@ -24,9 +24,9 @@
 
 def test_smoke():
     print("hello")
-    options = sdk_module.OpenmldbSdkOptions(case_conf.FEDB_ZK_CLUSTER,
+    options = sdk_module.OpenmldbClusterSdkOptions(case_conf.FEDB_ZK_CLUSTER,
                                    case_conf.FEDB_ZK_PATH)
-    sdk = sdk_module.OpenmldbSdk(options)
+    sdk = sdk_module.OpenmldbSdk(options, True)
```
Comment (dl239) [inline python/openmldb/test/sql_smoke_test.py]: `FEDB`->`OpenMLDB`

**Comment by HuilinWu2** [comment]
> > now users are able to use standalone mode in python SDK DBAPI through:
> > ```
> > import openmldb
> > # cluster mode
> > db = openmldb.dbapi.connect("db_test","127.0.0.1:6181","/onebox")
> > 
> > # standalone mode
> > db = openmldb.dbapi.connect("db_test", host="127.0.0.1", port=6527, is_cluster_mode=False)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or through sqlalchemy API:
> > ```
> > import sqlalchemy as db
> > # cluster
> > engine = db.create_engine("openmldb:///db_test?zk=127.0.0.1:6181&zkPath=/onebox")
> > 
> > # standalone
> > engine = db.create_engine("openmldb:///db_test?host=127.0.0.1&port=6527", connect_args={'is_cluster_mode':False})
> > ```
> 
> why add `is_cluster_mode` option to host and port connect method. It's better to keep consist with CLI that zk as cluster and host/port as standalone
agree, we have also considered. However, then we may only judge whether users want to use standalone/cluster mode by the type of second param, namely port int, '/onebox string' .

**Comment by HuilinWu2** [comment]
> > now users are able to use standalone mode in python SDK DBAPI through:
> > ```
> > import openmldb
> > # cluster mode
> > db = openmldb.dbapi.connect("db_test","127.0.0.1:6181","/onebox")
> > 
> > # standalone mode
> > db = openmldb.dbapi.connect("db_test", host="127.0.0.1", port=6527, is_cluster_mode=False)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or through sqlalchemy API:
> > ```
> > import sqlalchemy as db
> > # cluster
> > engine = db.create_engine("openmldb:///db_test?zk=127.0.0.1:6181&zkPath=/onebox")
> > 
> > # standalone
> > engine = db.create_engine("openmldb:///db_test?host=127.0.0.1&port=6527", connect_args={'is_cluster_mode':False})
> > ```
> 
> why add `is_cluster_mode` option to host and port connect method. It's better to keep consist with CLI that zk as cluster and host/port as standalone



RE: I agree, we have also considered. However, if so , then we may know whether users want to use standalone/cluster mode only by identifying the type of the second input, namely `zkPath string` or `port int`, do you agree?

File: python/openmldb/test/sql_smoke_test.py (hunk 5 above)
Comment (HuilinWu2) [reply python/openmldb/test/sql_smoke_test.py]: updated.

File: python/openmldb/test/case_conf.py [hunk 6]
```diff
@@ -19,5 +19,5 @@
 
 """
 
-FEDB_ZK_CLUSTER="127.0.0.1:6181"
-FEDB_ZK_PATH="/onebox"
+OpenMLDB_ZK_CLUSTER="127.0.0.1:6181"
+OpebMLDB_ZK_PATH="/onebox"
```
Comment (dl239) [inline python/openmldb/test/case_conf.py]: `Opeb`->`Open`

Comment (HuilinWu2) [reply python/openmldb/test/case_conf.py]: updated.

**Review by dl239 — APPROVED** [review]

**Pull request merged**